ocaml-cohttp-lwt/test/test_ocaml_lwt.ml

```diff
@@ -0,0 +1,141 @@
+open Alcotest
+open Ocaml_cohttp_lwt
+
+module Docker = struct
+  let container_name = "easyracer_server"
+
+  let run_docker_command cmd =
+    let open Unix in
+    let in_channel, out_channel, err_channel =
+      open_process_full cmd (environment ())
+    in
+    let output = ref "" in
+    let error = ref "" in
+
+    (try
+       while true do
+         output := !output ^ input_line in_channel ^ "\n"
+       done
+     with End_of_file -> ());
+
+    (try
+       while true do
+         error := !error ^ input_line err_channel ^ "\n"
+       done
+     with End_of_file -> ());
+
+    let exit_status =
+      close_process_full (in_channel, out_channel, err_channel)
+    in
+    match exit_status with
+    | WEXITED code ->
+        if code <> 0 then
+          failwith
+          @@ Printf.sprintf "Command failed with exit code: %d\nError: %s" code
+               !error
+        else !output
+    | WSIGNALED _ | WSTOPPED _ ->
+        failwith "Command was killed or stopped unexpectedly"
+
+  (* Start the Docker container and expose a port *)
+  let start_container port =
+    let image_name = "ghcr.io/jamesward/easyracer:latest" in
+    let docker_run_cmd =
+      Printf.sprintf "docker run --rm -d -p %d:%d --name %s %s --debug" port
+        port container_name image_name
+    in
+    let s = run_docker_command docker_run_cmd in
+    print_endline s;
+    (* Wait for 2 seconds, for simplicity *)
+    Unix.sleep 2
+
+  (* Stop and remove the Docker container *)
+  let stop_container () =
+    let docker_stop_cmd =
+      Printf.sprintf "docker kill -s SIGKILL %s" container_name
+    in
+    let s = run_docker_command docker_stop_cmd in
+    print_endline s
+end
+
+let port = 8080
+let right = "right"
+
+let test_scenario_1 () =
+  let response = Scenarios.scenario_1 port in
+  check string "Scenario 1 response" right response
+
+let test_scenario_2 () =
+  let response = Scenarios.scenario_2 port in
+  check string "Scenario 2 response" right response
+
+let test_scenario_3 () =
+  (* FIXME: Failing in GitHub Actions *)
```

Comment:
Do you know why this is failing on CI?  Do we need to bump the open files?

Reply:
It turns out, I had conf-libev installed in my local opam switch where I was developing. Using a fresh switch helped me debug this, and I've now fixed the dependencies so that libev (instead of select) is used in CI too.